test/python/transpiler/test_consolidate_blocks.py

```diff
@@ -655,15 +655,18 @@ def test_non_cx_target(self):
         self.assertEqual(Operator.from_circuit(qc), Operator(res.data[0].operation.params[0]))
 
     @data(["rzz", "rx", "rz"], ["rzz", "rx", "rz", "cz"])
-    def test_collect_rzz(self, basis_gates):
-        """Collect blocks with RZZ gates."""
+    def test_collect_and_synthesize_rzz(self, basis_gates):
+        """Collect blocks with RZZ gates, and re-synthesizing it."""
         qc = QuantumCircuit(2)
         qc.rzz(0.1, 0, 1)
         qc.rzz(0.2, 0, 1)
         consolidate_pass = ConsolidateBlocks(basis_gates=basis_gates)
         res = consolidate_pass(qc)
         self.assertEqual({"unitary": 1}, res.count_ops())
         self.assertEqual(Operator.from_circuit(qc), Operator(res.data[0].operation.params[0]))
+        pm = generate_preset_pass_manager(optimization_level=2, basis_gates=basis_gates)
+        tqc = pm.run(qc)
+        self.assertEqual(tqc.count_ops()["rzz"], 1)
```

Comment:
```suggestion
        # regression test for https://github.com/Qiskit/qiskit/issues/13428#issuecomment-2699855434
        pm = generate_preset_pass_manager(optimization_level=2, basis_gates=basis_gates)
        tqc = pm.run(qc)
        self.assertEqual(tqc.count_ops()["rzz"], 1)
```

Reply:
I added a reference to the issue, since all the discussion there is relevant.
Following your suggestion, I also extended `test_unitary_synthesis.py` with some tests that have basis_gates and target that include both RZZ and CZ gates.